modules/auxiliary/scanner/http/trace.rb

```diff
@@ -27,33 +25,82 @@ def initialize
           ['URL', 'https://www.owasp.org/index.php/Cross_Site_Tracing']
         ]
     )
+    register_options(
+      [
+        OptString.new('PATH', [ true, "The PATH to use while testing", '/']),
+        OptInt.new('TIMEOUT', [true, 'The socket connect/read timeout in seconds', 20]),
+        OptBool.new('COOKIE_CHECK', [ false, "Check for Cookie header also", false ])
+      ]
+    )
   end
 
   def run_host(target_host)
 
+    timeout = datastore['TIMEOUT']
+
+    rand_payload = Rex::Text.rand_text_alpha(5 + rand(8))
+
+    web_path = normalize_uri(datastore['PATH'])
+    xst_payload = "<script>alert(#{rand_payload})</script>" # XST Payload
+    check_uri = web_path + xst_payload
+
     begin
-      res = send_request_raw({
-        'uri'          => '/<script>alert(1337)</script>', #XST Payload
-        'method'       => 'TRACE',
-      })
+
+      if datastore['COOKIE_CHECK']
+        vprint_status("Sending request #{rhost}:#{rport} (vhost: #{vhost}) with Cookie header check")
+        res = send_request_raw({
+          'uri'    => check_uri,
+          'method' => 'TRACE',
+          'headers' => {
+            'Cookie' => "name=#{xst_payload}"
+          },
+        }, timeout)
+      else
+        vprint_status("Sending request #{rhost}:#{rport} (vhost: #{vhost})")
+        res = send_request_raw({
+          'uri'    => check_uri,
+          'method' => 'TRACE',
+        }, timeout)
+      end
 
       unless res
-        vprint_error("#{rhost}:#{rport} did not reply to our request")
+        vprint_error("#{rhost}:#{rport} (vhost: #{vhost})[#{res.code}] did not reply to our request")
```

Comment:
`res.code` is inside the `unless res` block here so I think it will always raise here